### PR TITLE
chore: Change to Call fetch snapshot API call only from edit mode

### LINKED ISSUE #24715: AxiosError: Request failed with status code 404
Sentry Issue: [APPSMITH-6CZ](https://appsmith.sentry.io/issues/4182077380/?referrer=github_integration)

```
AxiosError: Request failed with status code 404
```

### PR DESCRIPTION
**Pull request by rahulramesha**
## Description
This PR contains changes to call fetch snapshot API only in Edit mode

Fixes #24715 

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/SnapshotSagas.ts [hunk 1]
```diff
@@ -195,8 +195,7 @@ export default function* snapshotSagas() {
     ),
     takeLatest(
       [
-        ReduxActionTypes.INIT_CANVAS_LAYOUT,
```
Comment (marks0351) [inline app/client/src/sagas/SnapshotSagas.ts:198]: init canvas layout is removed, is that intentional?

**Review by marks0351 — APPROVED** [review]

File: app/client/src/sagas/SnapshotSagas.ts (hunk 1 above)
Comment (rahulramesha) [reply app/client/src/sagas/SnapshotSagas.ts:198]: @marks0351 yes, this is intentional. It was being called in both edit mode and view mode..

**Comment by rahulramesha** [comment]
@kamakshibhat-appsmith @Sripriya93 this needs some sanity on the snapshot flow. as i have changed the logic of making the backend call.. Things that specifically need to sanity are, snapshot banner should exist on page load and should discard once discarded..

**Comment by kamakshibhat-appsmith** [comment]
Looks good, we can merge @rahulramesha

**Pull request merged**